turbopack/crates/turbopack-ecmascript/src/references/esm/base.rs

```diff
@@ -161,8 +162,26 @@ impl ModuleReference for EsmAssetReference {
         }
         let ty = if matches!(self.annotations.module_type(), Some("json")) {
             EcmaScriptModulesReferenceSubType::ImportWithType(ImportWithType::Json)
-        } else if let Some(part) = &self.export_name {
-            EcmaScriptModulesReferenceSubType::ImportPart(*part)
+        } else if let Some(part) = self.export_name {
+            // This is a strange place to handle this, but see https://github.com/vercel/next.js/pull/70336
+            if *part.await? == ModulePart::Evaluation {
+                if let Some(module) = Vc::try_resolve_sidecast::<
+                    Box<dyn crate::EcmascriptChunkPlaceable>,
+                >(self.origin)
+                .await?
+                {
+                    if *module
+                        .is_marked_as_side_effect_free(
+                            self.origin.asset_context().side_effect_free_packages(),
+                        )
+                        .await?
+                    {
+                        return Ok(ModuleResolveResult::ignored().cell());
+                    }
+                }
+            }
+
```

Comment:
You don't want to add such logic here, but in `turbopack/crates/turbopack/src/lib.rs` instead. It's part of the "processing" to decide how to handle `ModulePart`.